pkg/endpoints/category/associate.go

```diff
@@ -0,0 +1,48 @@
+package category
+
+import (
+	"fmt"
+	"github.com/discmonkey/retext/pkg/db"
+	"log"
+	"net/http"
+)
+
+func AssociateEndpoint(store db.Store) func(w http.ResponseWriter, r *http.Request) {
+	t := func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodPost {
+			http.Redirect(w, r, "/", http.StatusSeeOther)
+			return
+		}
+
+		categoryID := r.FormValue("category")
+		documentID := r.FormValue("key")
+		text := r.FormValue("text")
+
+		if len(categoryID) == 0 {
+			w.WriteHeader(400)
+			log.Println("category parameter required")
+			return
+		}
+		if len(documentID) == 0 {
+			w.WriteHeader(400)
+			log.Println("key parameter required")
+			return
+		}
+		if len(text) == 0 {
+			w.WriteHeader(400)
+			log.Println("text parameter required")
+			return
+		}
+		err := store.CategorizeText(categoryID, documentID, text)
+
+		if err != nil {
```

Comment:
probably return another 400 here, maybe move this type of error handling to a function (and I know I did the same thing on my stuff, i'll follow your lead and correct it).

Reply:
where would an error handling function go if we're trying to share it between packages? make a util package in endpoints/?

Reply:
Maybe something like endpoints/common -- 
Not great, aka just barely better than utils, but also better than nothing